File: 202312/1716.go
func totalMoney(n int) int {
	ans := 0
	one := 0
	for i := 1; i <= n; i++ {
		if (i-1)%7 == 0 {
			one++
		}
		ans += one + (i-1)%7
	}
	return ans
}

File: 202510/1716.py
class Solution:
    def totalMoney(self, n: int) -> int:
        weeks = n // 7
        days = n % 7
        ans = 28 * weeks + weeks * (weeks - 1) // 2 * 7
        if days != 0:
            ans += (1 + weeks + weeks + days) * days // 2
        return ans
